### ISSUE TITLE
Why there is no option to prevent component rendering untill all observable variables are resolved.

### ISSUE THREAD

**Issue by sparkforreg** [angular/angular#6674]
I think it is worth to add option to disable change detection/rendering for component until component is fully loaded (all async http requests are executed)

**Comment by gdi2290** [comment]
The question can mean many thing and in some cases it's something you manage outside of the framework.

A. You can scan all observables before you bootstrap

B. Bootstrap then scan all observables before passing the object to the top level component.

C. You can also change the changeDetection within the component to either trigger when inputs changes or manually trigger the change detector them.

D. If you're using `|async` then it should only be used at the top level if you don't feel like using `.subscribe` but you really should just use `.subscribe`. 

E. If you're using `|async` everywhere then what you're really doing is inverting control over rendering to the observables which means you either need to do C, D, B, or A

**Comment by sparkforreg** [comment]
Thanks for response. In point C you said "or manually trigger the change detector them", so is it possible to disable default change detection trigger for particular component and trigger it manually? If it is possible can you provide documentation or example how to achieve it.

**Comment by gdi2290** [comment]
[ChangeDetectionStrategy](https://angular.io/docs/ts/latest/api/core/ChangeDetectionStrategy-enum.html)  doesn't seem to be working. You would simple set the component as `Detached`.

We can also use `ngOnInit` lifecycle hook to remove the component from the change detection tree. You would need to run `this.ref.detach();` where `ref` is injected via `ChangeDetectorRef`
```typescript
  ngOnInit() {
    this.ref.detach();
  }
  makeYourChanges() {
    this.ref.reattach(); // attach back to change detector tree

    this.data.value = Math.random() + ''; // make changes

    this.ref.detectChanges(); // check as dirty

    this.ref.detach(); // remove from tree
    // zone.js triggers changes
  }
```
[ChangeDetectorRef](https://angular.io/docs/ts/latest/api/core/ChangeDetectorRef-class.html)

You can also not include zone.js and manually control all changes. You can also inject `NgZone` to run an operation outside of zone.js so it won't tell angular to trigger chanes. For example,
```typescript
// this example might need a refactor to work with rxjs 5
export class Timeflies {
  pos   = 'absolute';
  color = 'red';
  letters: LetterConfig[];
  constructor(
    private service: Message,
    private el: ElementRef,
    private zone: NgZone) {

  }
  ngOnInit() {
    // initial mapping (before mouse moves)
    this.letters = this.service.message.map(
      (val, idx) => ({
        text: val,
        top: 100,
        left: (idx * 20 + 50),
        index: idx
      })
    );
    this.zone.runOutsideAngular(() => {
      Observable
        .fromEvent(this.el.nativeElement, 'mousemove')
        .map((e: MouseEvent) => {
          //var offset = getOffset(this.el);

          // subtract offset of the element
          var o = this.el.nativeElement.getBoundingClientRect();

          return {
            offsetX: e.clientX - o.left,
            offsetY: e.clientY - o.top
          };
        })
        .mergeMap(delta => {
          return Observable
            .fromArray(this.letters
              .map((val, index) => ({
                letter: val.text,
                delta,
                index
              })));
        })
        .mergeMap(letterConfig => {
          return Observable
            .timer( (letterConfig.index + 1) * 100)
            .map(() => ({
              text:  letterConfig.letter,
              top:   letterConfig.delta.offsetY,
              left:  letterConfig.delta.offsetX + letterConfig.index * 20 + 20,
              index: letterConfig.index
            }));
        })
        .subscribe(letterConfig => {
          // to render the letters, put them back into app zone
          this.zone.run(() => this.letters[letterConfig.index] = letterConfig);
        });

    });//zone
  }
}
```

**Comment by sparkforreg** [comment]
Thanks a lot. Your example with ChangeDetectorRef works for me.

**Comment by pkozlowski-opensource** [comment]
Sounds like a support question that got resolved (thnx @gdi2290 !). Closing.
